lib/web_ui/flutter_js/BUILD.gn

```diff
@@ -18,6 +20,17 @@ copy("flutter_js_sources") {
       [ "$root_out_dir/flutter_web_sdk/flutter_js/{{source_target_relative}}" ]
 }
 
+generated_file("flutter_js_engine_version") {
+  outputs = [ "$root_out_dir/flutter_web_sdk/flutter_js/src/revision.js" ]
+  contents = [
+    "// Copyright 2013 The Flutter Authors. All rights reserved.",
+    "// Use of this source code is governed by a BSD-style license that can be",
+    "// found in the LICENSE file.",
+    "",
+    "export const engineRevision = \"$engine_version\";",
```

Comment:
This scares me for a couple of reasons:

* The output of web build rules depends on every byte in the engine codebase. For example, changing a `README` file will result in altering the git revision and invalidating the build.
* Git revision does not fully describe the engine revision. You can have local source changes while the git revision is the same.

This probably invalidates one of the core principles of a build system, but I'm failing to name one, maybe hermeticity.

Reply:
Yeah this is an interesting one, I get your concerns. Could we maybe have a GVC and chat about the subtleties here at some point if you have a moment?

Reply:
We had some discussion in a GVC. I think I mostly agree that it would be better to leave the engine revision out of `flutter.js` and instead have the tool inject the engine hash as part of the build config.